src/command/run.rs

```diff
@@ -26,31 +26,28 @@ pub fn run(target: Target, in_file: PathBuf) -> Result<(), String> {
     build::build_to_buffer(target, &in_file, &mut buf)?;
     match target {
         Target::JS => {
-            let process = match Command::new("node")
+            let process = Command::new("node")
                 .stdin(Stdio::piped())
                 .stdout(Stdio::piped())
                 .stderr(Stdio::piped())
                 .spawn()
-            {
-                Err(why) => panic!("couldn't spawn wc: {}", why),
-                Ok(process) => process,
-            };
+                .map_err(|e| format!("Could not spawn Node.js process: {}", e.to_string()))?;
 
             process
                 .stdin
                 .unwrap()
                 .write_all(&buf)
-                .expect("Could not write to nodejs process");
+                .map_err(|e| format!("Could not write to Node.js process: {}", e.to_string()))?;
```

Comment:
```suggestion
                .map_err(|e| {
                    format!("Could not write to Node.js process: {}", e);
                    std::process::exit(1);
                })?;
```

Reply:
What would be the benefit of that? The error will be propagated upwards through the result, which will already cause the program to fail with a nice error message.

Reply:
Status codes. If program didn't finish successfully, the message is not enough, you also need to return non-0 exit code (usually 1 or 2). Shell and scripts usually take advantage of that:

- My prompt turns red if previous command failed
- Makefile recipe fails if command exits with non-0 code
- CI will also use that

Reply:
Nevermind